modules/packages/Curl.chpl

```diff
@@ -265,8 +265,9 @@ module Curl {
         var tmp:c_ptr(void) = arg.list:c_ptr(void);
         err = curl_easy_setopt_ptr(curl, opt:CURLoption, tmp);
       } else if arg.type == string || arg.type == bytes {
+        const localArg = arg.localize();
         err = curl_easy_setopt_ptr(curl, opt:CURLoption,
-                                   arg.localize().c_str():c_ptr(void));
+                                   localArg.c_str():c_ptr(void));
```

Comment:
I'm not understanding why this change is necessary, but I see it in a lot of places. It doesn't seem like something that should be needed since we have the unstable `.c_str()` still.

Reply:
changed all these based on conversation in https://github.com/chapel-lang/chapel/pull/22622#discussion_r1283627863.